test/Microsoft.AspNet.Mvc.Core.Test/UrlHelperTest.cs

```diff
@@ -807,6 +809,83 @@ public void LinkWithCustomHostAndProtocol_ReturnsExpectedResult()
             Assert.Equal("https://myhost/named/home/newaction/someid", url);
         }
 
+        // Regression test for aspnet/Mvc#2859
```

Comment:
IMO it's much more useful than a TODO. The point is that if you want to understand **all** of the context about these (odd) tests, you can find it.